etcdserver/server_test.go

```diff
@@ -939,8 +939,8 @@ func TestTriggerSnap(t *testing.T) {
 		srv.Do(context.Background(), pb.Request{Method: "PUT"})
 	}
 
-	srv.Stop()
 	<-donec
+	srv.Stop()
```

Comment:
`goAttach` of running snapshot operation exits (not executing the snapshot) from `EtcdServer.Stop` that is called before this line https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L1390. So the test was blocking forever to wait for snapshot to finish.

I think we should wait for snapshot to finish first (`<-donec`) and then stop the server?